### ISSUE TITLE
Great progress! But the project isn't working right now, is it?

### ISSUE THREAD

**Issue by shanmu-raoyunfei** [hughlv/agentok#134]
There is a lot of progress on the project right now, and you can see a lot of improvements. At present, the online version does not work properly. For example, when viewing the python code, it will give an error, and it can not reply to the problems such as running (and I do not see the location of the openai key). It does not work for the time being, right?

<img width="1594" alt="image" src="https://github.com/user-attachments/assets/74e88bf2-b928-4442-8d5a-a7ee5f228beb">

**Comment by hughlv** [comment]
This is expected behavior. You can take a look at the few templates https://studio.agentok.ai/marketplace I published recently for reference. (This should be mentioned in document, or as part of better UX. Not a priority to me for now.)

As a complete workflow, the first node should be Initializer, the second one can be any ConversableAgent, or User node, otherwise the workflow cannot be turned into a workable python program.

However, even though I think the basic structure should be fine now, I'm still considering the value of AutoGen, later on will iterate based on actual scenarios. If you're a programmer, any contributions are welcome.

**Comment by hughlv** [comment]
Today I did a simple patch to avoid your situation: When creating a new project, a default working workflow will be provided as sample for further editing.

**Comment by shanmu-raoyunfei** [comment]
Many thanks

**Comment by shanmu-raoyunfei** [comment]
Why does the demo in any of the markets I selected not work? In addition, there is no deletion in the project I fork out?

**Comment by hughlv** [comment]
> Why does the demo in any of the markets I selected not work? In addition, there is no deletion in the project I fork out?

No worries. I'm currently spending all my spare time doing a huge architecture refactoring. Will be back soon.

**Comment by shanmu-raoyunfei** [comment]
Great, because it's a great project, so keep an eye on it and think about how it applies to production. Thank you for your efforts

**Comment by hughlv** [comment]
> Great, because it's a great project, so keep an eye on it and think about how it applies to production. Thank you for your efforts

Understood. I'm happy you like it. And it's the reason why I should spend more time on the improvements.

**Comment by shanmu-raoyunfei** [comment]
When will this great project be updated, thanks

**Comment by hughlv** [comment]
> When will this great project be updated, thanks

I'm facing some difficulties integrating the RAG feature of autogen. Will try to catch up as soon as I get some spare time.

**Comment by hughlv** [comment]
> When will this great project be updated, thanks

@shanmu-raoyunfei I saw your Feature Request on AutoGen about RAG https://github.com/microsoft/autogen/issues/1387. Have you managed to get RAG agent working? I'm stuck by their docs and examples, because all the example does not seems to solve real problems, just pass in some docs and vectorize, this is definitely no the real case. What I'm expecting is to use an existing external knowledge base directly.

**Comment by shanmu-raoyunfei** [comment]
To be honest, all RAGs are currently mostly vectorized and used later. I understand what you mean, but it would be acceptable for us to vectorize the knowledge base when we configure it for the agent. There is an autogen open source project that can use data sources directly RAG, you can refer to below. The name of this project is: agentcloud

**Comment by hughlv** [comment]
> To be honest, all RAGs are currently mostly vectorized and used later. I understand what you mean, but it would be acceptable for us to vectorize the knowledge base when we configure it for the agent. There is an autogen open source project that can use data sources directly RAG, you can refer to below. The name of this project is: agentcloud

I know this project. will check out in detail later. thx.

**Comment by shanmu-raoyunfei** [comment]
Very happy to see the project has made great progress, more and more look forward to!

**Comment by shanmu-raoyunfei** [comment]
In addition, is our architecture BaaS, and does it support calling workflow through external API?

**Comment by hughlv** [comment]
> In addition, is our architecture BaaS, and does it support calling workflow through external API?

Yes it supports API by design. you can take a look at https://api.agentok.ai/v1/docs, which contains all the exposed APIs and you can test out with API key.
